debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/AbstractRecordsProducerTest.java

```diff
@@ -525,8 +525,9 @@ private Schema hstoreMapSchema() {
     protected List<SchemaAndValueField> schemaAndValuesForBinTypes() {
         return Arrays.asList(new SchemaAndValueField("ba", Schema.OPTIONAL_BYTES_SCHEMA, ByteBuffer.wrap(new byte[]{ 1, 2, 3 })),
                 new SchemaAndValueField("bol", Schema.OPTIONAL_BOOLEAN_SCHEMA, false),
-                new SchemaAndValueField("bs", Bits.builder(2).optional().build(), new byte[]{ 3, 0 }), // bitsets get converted from two's complement
```

Comment:
Why is it this got changed?

Reply:
Thanks for the review!

This was updated due to PostgresValueConverter.convertBits now using byte arrays only (no longer using short/int/long) and how debezium-core/src/main/java/io/debezium/jdbc/JdbcValueConverters.convertBits handles byte arrays vs short/int/long.

In the case of short/int/long, JdbcValueConverters.convertBits returns an array with length equal to the data type's length (2 for short, 4 for int, 8 for long) no matter if all elements are significant or not.
In the case of byte arrays, JdbcValueConverters.convertBits returns the array as is (except to convert to little-endian when necessary).

A value of '11' in the database was translated to a short of 3 which would create an array of {0,3} with the 0 being insignificant. Now, it is translated directly to a byte array of {3}.

Please let me know if ive misunderstood anything. Thanks again

Reply:
Ok, understood. Yes, agreed that the leading 0 is non-significant and it makes sense to get rid of it. This made me realize though that there's a problem with the currently proposed change: it reverses the endian-ness of the produced data (i.e. the order of the byte array).

This becomes apparent when using some values longer than a byte (the previous test data is really lacking in that regard...). E.g. consider a value like `B'000000110000001000000001'` (bytes 3, 2 and 1). Before your change, it would have been emitted as `[1, 2, 3, 0]`. I.e. "little-endian", with least significant byte as the first. Whereas now, with your change, it's `[3, 2, 1, 0]`. It actually should be `[1, 2, 3]` (little-endian as before, insignificant 0's removed). This means the expected value for your new column should be `[0, 0, 0, 0, 0, 0, 0, -128]`.

Note the little-endian byte ordering seems a bit odd IMHO, but it's the existing behavior and it also is described as such in the connector docs. Hence we should stick to it, yet removing insignificant leading 0's.

Does that make sense?

@jpechane, if you could double-check what I wrote, that'd be much appreciated. Thanks!

Reply:
Thanks again for the review @gunnarmorling and nice catch. I have updated to use BitSet instead of BigInteger/byte arrays and added an additional test case.